lib/relative.date.js

```diff
@@ -7,13 +7,28 @@ var noop = function noop () {};
 
 module.exports = function RelativeDate (date) {
   if (!(this instanceof RelativeDate)) return new RelativeDate(date);
-  date = date ? new Date(date) : new Date();
 
+  var lastTimeout = null;
   var listener = noop;
+
   this.onTick = function onTick (newListener) {
     listener = newListener;
   };
 
+  this.setDate = function setDate (newDate) {
+    date = newDate ? new Date(newDate) : new Date();
+    clearTick();
+    tick();
+  };
+
+  function clearTick () {
```

Comment:
Any reason why not `this.clearTick = function clearTick () {…}`, instead of a separate attribution (line 30)?